ref/general/observability-monitoring.adoc

```diff
@@ -0,0 +1,65 @@
+= Observability and monitoring
+
+Observability helps you understand system behavior by processing and monitoring the data the system generates through metrics, logging, tracing, and health checks, to get the information you need to troubleshoot failure.
+
+In a microservices environment, your application needs to be functional as a whole, though individual service instances can start or stop to adjust for capacity requirements or to recover from failure.
+To monitor this fluid system each entity must provide appropriate data for problem detection.
+Metrics collect numerical information from the application and platform.
+Whereas, logs collect textual information.
+Tracing, on the other hand, captures the end-to-end flow of a request through the system.
+Health Checks help orchestrators perform automated actions to maintain overall system health.
+
+== Metrics
+
+Metrics are numerical measurements that are captured and processed to identify and respond to potential issues before they escalate and cause more serious problems.
+In distributed system metrics, three are factors are involved; producers, aggregators, and processors.
+
+Producers, for example, applications, can be directly involved in producing metrics. Also, other infrastructure might passively observe or actively instrument the application to produce metrics on its behalf.
+What happens next depends on the aggregator.
+
+In the case of aggregators, for example, Prometheus, metrics are transferred from the producer to the aggregator by way of either `push` or `pull` mechanisms.
+Some aggregators, expect the application to connect to the aggregator to transmit data.
+The connection information for the aggregator must be distributed to all of the application processes that are measured.
+Other aggregators periodically connect to a known endpoint to gather metrics data.
+This requires the producer to define and provide an endpoint that can be scraped, and for the aggregator to be told where the endpoints are.
+When used with Kubernetes, Prometheus can discover endpoints based on service annotations.
+
+Processors, for example, Grafana, uses all of the aggregated data.
+Services like Grafana process aggregated metrics for visualization by using dashboards.
+Grafana also supports alerts by rules that are stored and evaluated independently of the dashboard.
+
+== Logs
+
+Log messages are strings with contextual information that pertains to the state and activity of a microservice when the log entry is created.
+Logs are required to diagnose how and why services fail. They play a supporting role to metrics in monitoring application health.
+Handling log files requires more thought if a containerized application can't be configured to write logs to `standard out` or `standard err`.
```

Comment:
For "standard out" and "standard err", add nouns after them to avoid any confusion about what they are.